airgun/views/host.py

```diff
@@ -39,26 +40,61 @@ class TableActions(View):
 class PuppetClassParameterValue(Widget):
     """Represent value field for Puppet Class parameters table row from Host
     Parameters tab. That field can be interacted with as usual text input, but
-    also it can be overriden with new value
+    also it can be overridden with new value
     """
     ROOT = ".//div[@class='input-group']"
     value = TextInput(
-        locator=".//textarea[contains(@name, 'lookup_values_attributes')]")
-    override_parameter = Text(".//a[@data-tag='override']")
-    remove_override = Text(".//a[@data-tag='remove']")
+        locator=".//*[self::textarea or self::input][contains(@name, 'lookup_values_attributes')]")
+    override_button = Text(".//a[@data-tag='override']")
+    remove_override_button = Text(".//a[@data-tag='remove']")
+    hide_button = Text(".//a[contains(@class, 'btn-hide')]")
+
+    @property
+    def hidden(self):
+        """Return whether the variable is hidden"""
+        return 'masked-input' in self.browser.classes(self.value)
+
+    @property
+    def overridden(self):
+        """Return whether the variable is overridden, a variable is overridden if not disabled"""
+        return self.browser.get_attribute('disabled', self.value) is None
+
+    @property
+    def hidden_value(self):
+        return self.browser.get_attribute('data-hidden-value', self.value)
 
     def read(self):
-        return self.value.read()
+        """Return smart variable widget values"""
+        return {'value': self.value.read(), 'overridden': self.overridden, 'hidden': self.hidden,
+                'hidden_value': self.hidden_value}
 
     def fill(self, value):
-        return self.value.fill(value)
+        """Set smart variable widget values"""
+        overridden = None
+        hidden = None
+        if isinstance(value, dict):
```

Comment:
Think the code is clear about the intention, but if you think that we should pass only dict I will update this code.

Reply:
Personally i like this implementation (in this particular case) - you can use both `my_parameter.fill('new_value')` in most cases or specify extra things if needed like `my_parameter.fill({'value': 'new_value', 'hidden': True})`.
The only better way i can see is different signature - `def fill(value, hidden=None, overridden=None)`, but i've looked through the code and 99% of our fill methods work with `value` only (in fact, we have only 1 exclusion), so maybe it's better to not introduce exclusion and follow the same way to avoid confusion.

Reply:
I don't really have strong opinion here. It's just everywhere where I looked, we pass dicts. So it seemed to me like doing extra work to cater for use-case that we never actually use. But I haven't looked everywhere and I might be missing something.

Let's leave it as is.